visualizations/frontend/filters/v2/TDatePicker.vue

```diff
@@ -0,0 +1,282 @@
+            }
+            this.selectedPreset = preset;
+        },
+        handleRangeDates(preset) {
+            let startDate = new Date();
+            let endDate = new Date();
+
+            switch (preset) {
+                case "last7days":
+                    startDate = new Date(startDate.setDate(startDate.getDate() - 7));
+                    endDate = new Date(endDate.setDate(endDate.getDate() - 1));
+                    break;
+                
+                case "lastmonth":
+                    const lastMonth = startDate.getMonth() - 1;
+                    startDate.setMonth(lastMonth);
+                    startDate.setDate(1); // First day of month
+                    endDate.setMonth(lastMonth);
+
+                    const lastDayOfMonth = new Date(endDate.getFullYear(), endDate.getMonth()+1, 0).getDate();
+                    endDate.setDate(lastDayOfMonth); // Last day of month
+                    break;
+                
+                case "last30days":
+                    startDate = new Date(startDate.setDate(startDate.getDate() - 30));
+                    endDate = new Date(endDate.setDate(endDate.getDate() - 1));
+                    break;
+                
+                case "last90days":
+                    startDate = new Date(startDate.setDate(startDate.getDate() - 90));
+                    endDate = new Date(endDate.setDate(endDate.getDate() - 1));
+                    break;
+                
+                case "mtd":
+                    startDate.setDate(1);
+                    break;
+                
+                case "ytd":
+                    startDate.setMonth(0);
+                    startDate.setDate(1);
+                    break;
+            }
+
+            this.date = [startDate, endDate];
+        },
+        handleSingleDates(preset) {
+            let date = new Date();
+            switch(preset) {
+                case "yesterday":
+                    date.setDate(date.getDate() - 1);
+                    break;
+
+                case "7DaysAgo":
+                    date.setDate(date.getDate() - 7);
+                    break;
+
+                case "30DaysAgo":
+                    date.setDate(date.getDate() - 30);
+                    break;
+                
+                case "90DaysAgo":
+                    date.setDate(date.getDate() - 90);
+                    break;
+                
+                case "yearAgo":
+                    date.setDate(date.getDate() - 365);
+                    break;
+                
+            }
+            this.date = date;
+        },
+        handleClear() {
+            this.date = [];
+            this.selectedPreset = {};
+            this.unsetFilterValue("start_date");
+            this.unsetFilterValue("end_date");
+            this.unsetFilterValue("date_preset");
+            this.unsetFilterValue("date");
+        },
+        formatDate(dateValue) {
```

Comment:
Do we have access to `moment` here? It might be helpful to use it for formatting dates.